### PR TITLE
migrate to slog

### PR DESCRIPTION
**Pull request by shashwatj07**
See https://github.com/tikv/client-rust/pull/294#issuecomment-849344011

### CODE REVIEW

**Comment by ekexium** [comment]
I think we should let the user pass the logger when creating a Client.
It may accept an `Option<Logger>` and create a default one if it's `None`.

**Comment by shashwatj07** [comment]
> I think we should let the user pass the logger when creating a Client.
> It may accept an `Option<Logger>`. We can create a default one if it's `None`.

@ekexium done, please take a look

File: src/raw/client.rs [hunk 1]
```diff
@@ -67,15 +70,22 @@ impl Client {
...
     /// # });
     /// ```
     pub async fn new_with_config<S: Into<String>>(
         pd_endpoints: Vec<S>,
         config: Config,
+        optional_logger: Option<Logger>,
     ) -> Result<Client> {
-        let plain = slog_term::PlainSyncDecorator::new(std::io::stdout());
-        let logger = Logger::root(slog_term::FullFormat::new(plain).build().fuse(), o!());
+        let logger = match optional_logger {
```
Comment (ekexium) [inline src/raw/client.rs:82]: Can be simplified as `unwrap_or_else`

File: src/transaction/client.rs [hunk 2]
```diff
@@ -75,15 +78,22 @@ impl Client {
...
     /// # });
     /// ```
     pub async fn new_with_config<S: Into<String>>(
         pd_endpoints: Vec<S>,
         config: Config,
+        optional_logger: Option<Logger>,
     ) -> Result<Client> {
-        let plain = slog_term::PlainSyncDecorator::new(std::io::stdout());
-        let logger = Logger::root(slog_term::FullFormat::new(plain).build().fuse(), o!());
+        let logger = match optional_logger {
```
Comment (ekexium) [inline src/transaction/client.rs:90]: Same here

**Review by ekexium — COMMENTED** [review]
Just a nitpick, mostly LGTM. Thanks!
Please fix the CI and the `examples`.

File: src/raw/client.rs (hunk 1 above)
Comment (shashwatj07) [reply src/raw/client.rs]: done

File: src/transaction/client.rs (hunk 2 above)
Comment (shashwatj07) [reply src/transaction/client.rs]: done

**Comment by shashwatj07** [comment]
@ekexium ping

**Review by ekexium — APPROVED** [review]

**Pull request merged**